viz.go

```diff
@@ -0,0 +1,147 @@
+package dataset
+
+import (
+	"io"
+	"encoding/json"
+	"fmt"
+
+	"github.com/ipfs/go-datastore"
+)
+
+// Viz stores configuration data related to representing a dataset as a
+// visualization
+type Viz struct {
+	// private storage for reference to this object
+	path datastore.Key
+	// Qri should always be "vc:0"
+	Qri Kind
+	// Format designates the visualization configuration syntax. currently the
+	// only supported syntax is "html"
+	Format string
+	// Script is a reader of raw script data
+	Script io.Reader `json:"_"`
+	// ScriptPath is the path to the script that created this
+	ScriptPath string `json:"script,omitempty"`
+}
+
+// Path gives the internal path reference for this structure
+func (v *Viz) Path() datastore.Key {
+	return v.path
+}
+
+// NewVizRef creates an empty struct with it's internal path set
+func NewVizRef(path datastore.Key) *Viz {
+	return &Viz{path: path}
+}
+
+// IsEmpty checks to see if Viz has any fields other than the internal path
+func (v *Viz) IsEmpty() bool {
+	return v.Format == "" && v.ScriptPath == ""
+}
+
+// SetPath sets the internal path property of a Viz
+// Use with caution. most callers should never need to call SetPath
+func (v *Viz) SetPath(path string) {
```

Comment:
Is this only used internally? If so, can it be renamed setPath.

Reply:
SetPath is used way up the stack: https://github.com/qri-io/qri/blob/abb36a425321576fe922ab72bdf2b59e8d212ce4/lib/datasets.go#L520

Once we fix that issue, we should be able to remove `SetPath` entirely, but for now, we'll need it to make an update that only affects `Viz` work properly